### PR TITLE
Mayhem Integration

### PR DESCRIPTION
**Pull request by code-terror**
## Motivation

(Write your motivation here)

## Test Plan

(Write your test plan here)

## Related PRs

(Link your related PRs here)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,59 +1,18 @@
-FROM ubuntu:18.04 AS builder
```
Comment (xansec) [inline Dockerfile]: Please don't modify upstream Dockerfiles.

File: Mayhem/deserialization.mayhemfile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+project: code-terror/snarkos
```
Comment (xansec) [inline Mayhem/deserialization.mayhemfile]: Please use generic project names.

File: mayhem.dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Build Stage
```
Comment (xansec) [inline mayhem.dockerfile:1]: Please don't use git clone.

**Review by xansec — CHANGES_REQUESTED** [review]
Please make the requested changes.

**Comment by code-terror** [comment]
@xansec Packaging this repository doesnt work, so using build image for analysis

**Comment by xansec** [comment]
> Packaging this repository doesnt work

What error are you encountering? I took your existing Dockerfile and installed libssl in the package stage, and it seems to work fine:
```
$ cat mayhem.dockerfile   

# Build Stage
FROM ubuntu:20.04 as builder

## Install build dependencies.
RUN apt-get update && \
    DEBIAN_FRONTEND=noninteractive apt-get install -y cmake clang curl git-all build-essential libssl-dev pkg-config
RUN curl --proto "=https" --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
RUN ${HOME}/.cargo/bin/rustup default nightly
RUN ${HOME}/.cargo/bin/cargo install -f cargo-fuzz
ADD . /snarkOS
WORKDIR /snarkOS/fuzz/
RUN ${HOME}/.cargo/bin/cargo fuzz build
WORKDIR /

FROM ubuntu:20.04
RUN apt-get update && apt-get install -y openssl
RUN mkdir -p /snarkOS/fuzz
COPY --from=builder /snarkOS/fuzz /snarkOS/fuzz
```

Can you try with this approach?

**Comment by code-terror** [comment]
@xansec Hmm...I was trying with libssl-dev and a bunch of errors were popping up..this approach works..

**Review by xansec — APPROVED** [review]
LGTM!

**Pull request merged**